server/api/crud/projects.py

```diff
@@ -215,18 +214,15 @@ async def list_project_summaries(
         state: mlrun.common.schemas.ProjectState = None,
         names: typing.Optional[list[str]] = None,
     ) -> mlrun.common.schemas.ProjectSummariesOutput:
-        projects_output = await fastapi.concurrency.run_in_threadpool(
-            self.list_projects,
-            session,
-            owner,
-            mlrun.common.formatters.ProjectFormat.name_only,
-            labels,
-            state,
-            names,
-        )
-        project_summaries = await self.generate_projects_summaries(
-            projects_output.projects
-        )
+        if not self._project_resource_counters_cache:
+            await self.refresh_project_resources_counters_cache(session)
+
+        project_summaries = []
+        for project_name in names or self._project_resource_counters_cache.keys():
+            project_summaries.append(
+                self._project_resource_counters_cache.get(project_name)
+            )
```

Comment:
Isn't it the same as?
```suggestion
        project_summaries = self._project_resource_counters_cache.values() or []
```

Reply:
no, since you can request specific names.
see 
```
for project_name in names or self._project_resource_counters_cache.keys():
```

Reply:
oh I was confused by the syntax. maybe use `names = names or self._project_resource_counters_cache.keys()` for readability.